webapp/components/backstage/installed_commands.jsx

```diff
@@ -21,16 +22,18 @@ export default class InstalledCommands extends React.Component {
         this.deleteCommand = this.deleteCommand.bind(this);
 
         this.state = {
-            commands: IntegrationStore.getCommands(),
-            loading: !IntegrationStore.hasReceivedCommands()
+            commands: IntegrationStore.getCommands(TeamStore.getCurrentId()),
+            loading: true
         };
     }
 
     componentDidMount() {
         IntegrationStore.addChangeListener(this.handleIntegrationChange);
 
-        if (window.mm_config.EnableCommands === 'true' && this.state.loading) {
-            AsyncClient.listTeamCommands();
+        if (window.mm_config.EnableCommands === 'true') {
+            AsyncClient.listTeamCommands(() => {
```

Comment:
This is the wrong way to go about this. You check if you have commands available not if you have loaded them or not. If you need some sort of loading you can set the commands to some nil value and check for that.